### ISSUE TITLE
false positive 'total-functions/no-unsafe-type-assertion' rule is disabled but never reported  eslint-comments/no-unused-disable

### ISSUE THREAD

**Issue by JustFly1984** [danielnixon/eslint-plugin-total-functions#38]
```
public getMessages(locale?: Locale): IntlMessages {
    if (locale !== undefined) {
      const messages = this.messages.get(locale)

      if (messages !== undefined) {
        return messages
      }
    }

    const messages = this.messages.get(this.locale)

    if (messages !== undefined) {
      return messages
    }

    // eslint-disable-next-line total-functions/no-unsafe-type-assertion
    return {} as IntlMessages
  }
```

Error "'total-functions/no-unsafe-type-assertion' rule is disabled but never reported"  doesn't show up in vscode, but shows up in terminal

Basically I need to return empty object at the end, so it has same type.

**Comment by danielnixon** [comment]
It sounds like your `rule is disabled but never reported` message is coming from eslint-plugin-eslint-comments (https://github.com/mysticatea/eslint-plugin-eslint-comments/pull/45/files).

What is the type of `IntlMessages`? It sounds like `no-unsafe-type-assertion` considers this a safe typecast, so is it as simple as removing the `eslint-disable-next-line` comment?

**Comment by danielnixon** [comment]
> so is it as simple as removing the eslint-disable-next-line comment?

And maybe even remove the type assertion itself too? So just `return {}`?

**Comment by JustFly1984** [comment]
@danielnixon removing eslint-disable-next-line provides correct error about unsafe type assertion. 

IntlMessages type is `{ [key: typeof keyof import('./messages-en.json')]: string } ` - basically flat keys of key/value pairs of dictionary.

The issue is not the code, issue is that in terminal there is an error about this rule is useless, then it is really useful.
But in vscode - no error.

**Comment by danielnixon** [comment]
Ah, so it sounds like `no-unsafe-type-assertion` is doing the right thing here and the problem is `eslint-plugin-eslint-comments`
